week-03/assessment/solar-farm/solar-farm/src/main/java/learn/solar/models/Panel.java

```diff
@@ -0,0 +1,113 @@
+package learn.solar.models;
+
+import java.util.Objects;
+
+//-- solar panel model
+public class Panel {
+    private int id;
+    private String section;
+    private int row;
+    private int column;
+    private int installationYear;
+    private Material material;
+    private boolean tracking;
+
+    //constructor
+    public Panel(){
+
+    }
+    public Panel(int id, String section, int row, int column, int installationYear, Material material, boolean tracking) {
+        this.id = id;
+        this.section = section;
+        this.row = row;
+        this.column = column;
+        this.installationYear = installationYear;
+        this.material = material;
+        this.tracking = tracking;
+    }
+
+    //full getters and setters
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public String getSection() {
+        return section;
+    }
+
+    public void setSection(String section) {
+        this.section = section;
+    }
+
+    public int getRow() {
+        return row;
+    }
+
+    public void setRow(int row) {
+        this.row = row;
+    }
+
+    public int getColumn() {
+        return column;
+    }
+
+    public void setColumn(int column) {
+        this.column = column;
+    }
+
+    public int getInstallationYear() {
+        return installationYear;
+    }
+
+    public void setInstallationYear(int installationYear) {
+        this.installationYear = installationYear;
+    }
+
+    public Material getMaterial() {
+        return material;
+    }
+
+    public void setMaterial(Material material) {
+        this.material = material;
+    }
+
+    public boolean isTracking() {
+        return tracking;
+    }
+
+    public void setTracking(boolean tracking) {
+        this.tracking = tracking;
+    }
+
+    // override `equals` and `hashCode`
+    //When we override the equals and the hashcode we are determining object equality
+    //so we can decide what makes two object equal
+    @Override
+    public String toString() {
+        return "Panel{" +
+                "id=" + id +
+                ", section='" + section + '\'' +
+                ", row=" + row +
+                ", column=" + column +
+                ", installationYear=" + installationYear +
+                ", material=" + material +
+                ", tracking=" + tracking +
+                '}';
+    }
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Panel panel = (Panel) o;
+        return id == panel.id && row == panel.row && column == panel.column && installationYear == panel.installationYear && tracking == panel.tracking && Objects.equals(section, panel.section) && material == panel.material;
```

Comment:
too many things in your equals method - we are only concerned with section, row and column here